numpyro/hmc_util.py

```diff
@@ -8,7 +8,8 @@
 from numpyro.util import cond, laxtuple, while_loop
 
 AdaptWindow = laxtuple("AdaptWindow", ["start", "end"])
-AdaptState = laxtuple("AdaptState", ["step_size", "inverse_mass_matrix", "ss_state", "mm_state", "window_idx"])
+AdaptState = laxtuple("AdaptState", ["step_size", "inverse_mass_matrix", "ss_state", "mm_state",
+                                     "window_idx", "rng"])
```

Comment:
I think it will be nice to have a consistent pattern for passing in `rng` to functions. Right now, `HMCState` and `AdaptState` take it as a last arg, and some other functions take it as a first arg. Shouldn't be part of this PR, but something that we can think about for future refactors.

Reply:
I only observe the inconsistency in build_tree functions. Will address it in a later PR. :)